### ISSUE TITLE
Can't import and use `next/image` in Dynamically imported MDX file in Next.js 13.3

### ISSUE THREAD

**Issue by iliaamiri** [vercel/next.js#48176]
### Summary

I'm making a personal blog using Next.js 13.3.0 and `@next/mdx` to integrate Markdown with Next.

I followed the docs to setup the basics and it seems to work just fine. BUT, when I try to dynamically import the MDX file using [Next's Dynamic Import](https://nextjs.org/docs/advanced-features/dynamic-import), I can no longer use `next/image` component in my MDX file.

---
## My MDX file looks like this:
```
import Image from "next/image";
import Image_Test from './assets/an-image.png';

`console.log("Hello world")`

The normal image tag works fine and I can see the markdown being compiled and loaded on the page with no issue.
<img src={Image_Test.src} /> 

But, this line breaks everything.
<Image src={Image_Test} width={900} height={200} className='w-full' />
```
---
## Parts of my page component file and file structure
```typescript
import dynamic from "next/dynamic";
import {useParams} from "next/navigation";

// ... other stuff

export default function BlogPost() { 
     // ... other stuff
     const params = useParams();

     const MDXBlogPost = dynamic(() => import(
        `@/app/blog/${params.slug}/post.mdx`
     ), {
        ssr: true,
     });

     return (
        <BlogPageWrapper coverImage={`/assets/blogs/${blog.slug}/post-cover.png`} blog={blog}>
            {/* @ts-ignore - dynamic import doesn't understand .MDX types */}
            <MDXBlogPost components={{code}} />
        </BlogPageWrapper>
     );
}
```

![image](https://user-images.githubusercontent.com/37903573/230802853-3e11c675-7999-401c-933d-b43c807194af.png)

## The Error I see on the page
![image](https://user-images.githubusercontent.com/37903573/230803041-ac15ddf2-0cac-4671-8132-baacde191a48.png)

When I remove that `<Image src ... />` line from Next Image, everything works like a charm. But, I want to use that component from Next.js

## My Device Info
- Windows 11
- Node v18

## Sources I've looked at to fix the problem
I have read these sources but I couldn't find any solution related to my problem.
- https://blog.yorunohikage.fr/2021/03/07/dynamically-imported-components-in-mdx-and-nextjs/
- https://github.com/vercel/next.js/discussions/12151

_Originally posted by @iliaamiri in https://github.com/vercel/next.js/discussions/48175_

**Comment by selven** [comment]
We have the same issue, this is the case for js files as well, it's not mdx only

**Comment by iliaamiri** [comment]
> We have the same issue, this is the case for js files as well, it's not mdx only

Yeah, it might be an issue of the bundler (in my case Webpack) doesn't include Next.js library when we dynamically import the files as it loses the track of the import. At least that's my guess of what's happening.

**Comment by realfresh** [comment]
I'm about to end my programming career from the amount of time I have wasted trying to get this to work

**Comment by iliaamiri** [comment]
@realfresh 

Yeah, I ended up just going with the regular `img` tag instead of `next/image`. 

I think if I just pass the `width` and `height` attributes to the normal `img` tag, I'm following `next/image`'s philosophy closer. But, still this is stupid.

**Comment by franky47** [comment]
My two cents: your page is a server component, and the Image component requires client-side hooks. Somehow the dynamic import loses the connection and doesn't show you the usual "this needs to be a client component" error.

What happens if you were to move the dynamic import to a file marked with `'use client'`?

**Comment by iliaamiri** [comment]
> My two cents: your page is a server component, and the Image component requires client-side hooks (context here, which ends up undefined).
> 
> Somehow the dynamic import loses the connection and doesn't show you the usual "this needs to be a client component" error.
> 
> What happens if you were to move the dynamic import to a file marked with `'use client'`?
> 
> The downside of going for a dynamic import is that the whole MDX file becomes a client component (ie: sent to the client as part of the JS bundle rather than just rendered to HTML/CSS).

I here you. The page is client component as well. I made sure of that. But it still does not work.

**Comment by iliaamiri** [comment]
Should we close this issue at this point? 

No one else seems to care
